tools/Az.Tools.Predictor/Az.Tools.Predictor/InterceptSurvey.ps1

```diff
@@ -0,0 +1,204 @@
+# ----------------------------------------------------------------------------------
+#
+# Copyright Microsoft Corporation
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+# http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.internal
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ----------------------------------------------------------------------------------
+
+param (
+    [Parameter(Mandatory)]
+    [string] $moduleName,
+    [Parameter(Mandatory)]
+    [int] $majorVersion
+)
+
+if ([string]::IsNullOrWhiteSpace($moduleName)) {
```

Comment:
This script isn't covering all the cases because this is a short term solution. There is a plan for a long term solution which still have some nuances.

Here are some cases that are not covering or may not doing well:
1. Having two modules using this intercept mechanism.
2. Intercept when the major version is changed.
3. Isn't sure if we need `surveyId` in future.

Reply:
Same question as to mark this or not